impl/src/main/java/com/sun/faces/util/Util.java

```diff
@@ -1482,6 +1484,19 @@ public static BeanManager getCdiBeanManager(FacesContext facesContext) {
             if (result == null && facesContext != null) {
                 Map<String, Object> applicationMap = facesContext.getExternalContext().getApplicationMap();
                 result = (BeanManager) applicationMap.get("org.jboss.weld.environment.servlet.jakarta.enterprise.inject.spi.BeanManager");
+
+                if (result == null && applicationMap.get("org.jboss.weld.environment.servlet.enhancedListenerUsed") == Boolean.TRUE) {
+                    LOGGER.log(WARNING, "Weld skipped initialization - forcing it to reinitialize");
+                    try {
+                        ServletContext servletContext = (ServletContext) facesContext.getExternalContext().getContext();
+                        servletContext.setInitParameter("org.jboss.weld.environment.servlet.archive.isolation", "false");
+                        ServletContainerInitializer weld = (ServletContainerInitializer) Class.forName("org.jboss.weld.environment.servlet.EnhancedListener").getConstructor().newInstance();
+                        weld.onStartup(null, servletContext);
+                        result = (BeanManager) applicationMap.get("org.jboss.weld.environment.servlet.jakarta.enterprise.inject.spi.BeanManager");
+                    } catch (Exception | LinkageError e) {
+                        LOGGER.log(WARNING, "Reinitializing Weld failed - giving up, please make sure your project has at least one @Named bean and retry", e);
```

Comment:
I don't think `@Named` is enough here.
If I understand correctly what you're running into, there are two factors in play. Firstly it's the discovery mode the app uses (`all` or `annotated`, the latter being default from CDI 4.0). Secondly, if it's `annotated`, then you need to have some bean with bean defining annotation[1][2]; `@Named` isn't bean defining.
I assume you're hitting this only with `annotated` discovery (since with `all` you're likely to find something that could be a bean), so you could log something like:

> `"Reinitializing Weld failed - giving up, please make sure your project contains at least one bean class with a bean defining annotation and retry"`

[1] https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0.html#bean_defining_annotations
[2] https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0.html#bean_defining_annotations_full

Reply:
Point taken, fixed.